### ISSUE TITLE
Fix S1075 FP: Escaped New Line Characters Detected as URL

### ISSUE THREAD

**Issue by eXpl0it3r** [SonarSource/sonar-dotnet#7785]
### Description

Using `\\r\\n` or similar as a string, triggers the rule S1075 despite `\\r\\n` not being paths.

Maybe this is by design, as `\\r` could be a network device, then again, I'm not sure if the specs define a naming limit.

### Repro steps

```csharp
public class CsvDialect
{
    public string Delimiter { get; set; }
    public string LineTerminator { get; set; }
}

// ...

var csvDialect = new CsvDialect
                     {
                         Delimiter = "|",
                         LineTerminator = @"\\r\\n"
                     }
```

### Expected behavior

No warning is issued

### Actual behavior

Warning is issued:

> Refactor your code not to use hardcoded absolute paths or URIs.

### Known workarounds

Looks like setting it as `const string` prevents the warning:

```csharp
const string LineTerminator = @"\\r\\n";
// ...
var csvDialect = new CsvDialect
                     {
                         Delimiter = "|",
                         LineTerminator = LineTerminator
                     }
```

**Comment by sebastien-marichal** [comment]
Hello @eXpl0it3r,

Thank you for your feedback.

I was not able to reproduce the issue with your code snippet.

How did you get this issue? Was it during a code analysis or with SonarLint?
Could you please provide more information about your environment?
Such as:
* C# Plugins version
* Visual Studio version (if used)
* SonarLint version (if used)
* MSBuild / dotnet version
* SonarScanner for .NET version (if used)
* Operating System

Thank you

**Comment by eXpl0it3r** [comment]
You're right, sorry about that. I moved the example out of a large project, so I didn't properly test it in isolation. To reproduce the issue, you have to construct the `CsvDialect` when creating another class. Here's an example that reproduced the issue for me in isolation:

```csharp
var file = new FileConfiguration
{
    Name = "File.txt",
    CsvDialect = new CsvDialect
    {
        Delimiter = "|",
        LineTerminator = @"\\r\\n"
    }
};

Console.WriteLine($"{file.Name}: {file.CsvDialect.Delimiter}+${file.CsvDialect.LineTerminator}");

public class CsvDialect
{
    public string Delimiter { get; set; }
    public string LineTerminator { get; set; }
}

public class FileConfiguration
{
    public string Name { get; set; }
    public CsvDialect CsvDialect { get; set; }
}
```

![image](https://github.com/SonarSource/sonar-dotnet/assets/920861/6d2f3f43-7bdc-4517-9a19-a1809e212339)


Metadata:

- Local
  - Windows 11
  - JetBrains Rider 2023.2
  - .NET 6 / 7
  - SonarAnalyzer.CSharp 9.7.0.75501
- CI
  - Ubuntu (Docker)
  - .NET 6
  - Whatever version `dotnet tool install --global dotnet-sonarscanner` currently installs

**Comment by sebastien-marichal** [comment]
Hello @eXpl0it3r,

Sorry for the delay.

I don't consider this as an FP. As you mentioned `@"\\r\\n"` could be a network path.
Technically, if you want to specify the line ending characters you could use `"\r\n"` or `Environment.NewLine`.

Maybe I am missing something from what you are trying to achieve here.

This issue could be marked as `Won't fix` in SonarQube or SonarCloud.

**Comment by eXpl0it3r** [comment]
`\r\n` and `\\r\\n` are not the same though. `\r\n` will be the literal characters codes for carriage return and line feed, while `\\r\\n` is the string representation where `\\` escapes `\`.

I'd say that `\\r`, `\\n`, `\\t` are not uncommon escaped characters to warrant an exclusion (FP), especially since one-letter network are a) very uncommon and b) [not necessarily allowed](https://serverfault.com/questions/162038/are-one-letter-host-names-valid).

If you want to keep it as is, what about the false negative then, where it's not detected unless it's a compound initialization?

For example, this doesn't trigger the error:
```csharp
var csvDialect = new CsvDialect
{
    Delimiter = "|",
    LineTerminator = @"\\r\\n"
};

Console.WriteLine($"{csvDialect .Delimiter}+${csvDialect .LineTerminator}");

public class CsvDialect
{
    public string Delimiter { get; set; }
    public string LineTerminator { get; set; }
}
```

**Comment by sebastien-marichal** [comment]
If I can add, `"\r\n"`, `"\\r\\n"` and `@"\\r\\n"` are not the same.
If you are looking to have the string representation, what about dropping the verbatim string?

```cs
var file = new FileConfiguration
{
    Name = "File.txt",
    CsvDialect = new CsvDialect
    {
        Delimiter = "|",
        LineTerminator = "\\r\\n" // <-- Non verbatim string, S1075 not raised
    }
};
```
`@"\\r\\n"` is not the string representation of the carriage return, `"\\r\\n"` is.

I hope this helps.

----

About your suggestion, it uncovers a potential FN:
```cs
var myObject = new MyObject
{
    FilePath = @"/some/path" // S1075 should raised
}
```

I opened an [issue](https://github.com/SonarSource/sonar-dotnet/issues/7815) to track it
